IntegrationTests/UITests/UITest.cs

```diff
@@ -38,18 +39,21 @@ void HandleApplicationIdle(object sender, EventArgs e)
             {
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                 await WaitForIdleAsync();
-                var dialog = Application.OpenForms.OfType<T>().Single();
+                var form = Application.OpenForms.OfType<T>().Single(); // is disposed finally by form.Close() below.
                 try
                 {
-                    await runAsync(dialog);
+                    Assert.True(form.Visible, $"{form.GetType().FullName} {form.Name} should be visible");
+                    await runTestAsync(form);
                 }
                 finally
                 {
-                    dialog.Close();
+                    form.Close();
+
+                    // Do not call form.Dispose() explicitly, because the FormClosed event would be called on the disposed form.
```

Comment:
💭 I thought we established earlier that we do want to call Dispose. Is that no longer the case?

Reply:
Your PRs -- I guess NUnit -- have changed the sequence of Windows events.
I tried it even with `Application.DoEvents` right after `form.Close`. Both with and without this call, the `OnFormClosed` was called after `Dispose`, which would be an unnatural behavior.

Reply:
Note that the call to Dispose would not appear _here_. It would appear indirectly by turning line 42 into a using declaration statement.

Reply:
If it's still closing the window in a strange order, it might mean the implementation of `showForm` is blocking on the completion of `runTestAsync` (e.g. by pushing a message loop) and somehow calling `Close` itself.

Reply:
If we can't do it in a more convenient location, we could call `Dispose` after we call `test.Join()` below.

Reply:
> Note that the call to Dispose would not appear here. It would appear indirectly by turning line 42 into a using declaration statement.

I have learned this meanwhile. That's why I have added a comment to line 42 that `Dispose` is called by `Close` (or by `Form.ShowDialog`).

> If it's still closing the window in a strange order, it might mean the implementation of `showForm` is blocking on the completion of `runTestAsync` (e.g. by pushing a message loop) and somehow calling `Close` itself.

Yes, this is how it felt debugging several variants.

> If we can't do it in a more convenient location, we could call `Dispose` after we call `test.Join()` below.

Will add this as a safety net.